spring-integration-core/src/main/java/org/springframework/integration/channel/FluxMessageChannel.java

```diff
@@ -43,55 +41,53 @@
 public class FluxMessageChannel extends AbstractMessageChannel
 		implements Publisher<Message<?>>, ReactiveStreamsSubscribableChannel {
 
-	private final List<Subscriber<? super Message<?>>> subscribers = new ArrayList<>();
+	private final AtomicInteger subscribed = new AtomicInteger();
 
-	private final Map<Publisher<? extends Message<?>>, ConnectableFlux<?>> publishers = new ConcurrentHashMap<>();
+	private final EmitterProcessor<Message<?>> flux;
 
-	private final Flux<Message<?>> flux;
+	private final EmitterProcessor<Integer> subscriptionDelay = EmitterProcessor.create(false);
 
 	private FluxSink<Message<?>> sink;
 
 	public FluxMessageChannel() {
-		this.flux =
-				Flux.<Message<?>>create(emitter -> this.sink = emitter, FluxSink.OverflowStrategy.IGNORE)
-						.publish()
-						.autoConnect();
+		this.flux = EmitterProcessor.create(1, false);
+		this.sink = this.flux.sink();
 	}
 
 	@Override
 	protected boolean doSend(Message<?> message, long timeout) {
-		Assert.state(this.subscribers.size() > 0,
+		Assert.state(this.subscribed.get() > 0,
 				() -> "The [" + this + "] doesn't have subscribers to accept messages");
 		this.sink.next(message);
 		return true;
 	}
 
 	@Override
 	public void subscribe(Subscriber<? super Message<?>> subscriber) {
-		this.subscribers.add(subscriber);
-
-		this.flux.doOnCancel(() -> this.subscribers.remove(subscriber))
+		this.flux.doFinally((signal) -> this.subscribed.decrementAndGet())
 				.retry()
 				.subscribe(subscriber);
-
-		this.publishers.values().forEach(ConnectableFlux::connect);
+		this.subscriptionDelay.onNext(this.subscribed.incrementAndGet());
 	}
 
 	@Override
 	public void subscribeTo(Publisher<? extends Message<?>> publisher) {
-		ConnectableFlux<?> connectableFlux =
-				Flux.from(publisher)
-						.handle((message, sink) -> sink.next(send(message)))
-						.onErrorContinue((throwable, event) ->
-								logger.warn("Error during processing event: " + event, throwable))
-						.doOnComplete(() -> this.publishers.remove(publisher))
-						.publish();
-
-		this.publishers.put(publisher, connectableFlux);
-
-		if (!this.subscribers.isEmpty()) {
-			connectableFlux.connect();
-		}
+		Flux.from(publisher)
+				.handle((message, sink) -> sink.next(send(message)))
+				.onErrorContinue((throwable, event) ->
+						logger.warn("Error during processing event: " + event, throwable))
+				.delaySubscription(
+						Mono.fromSupplier(this.subscribed::get)
+								.filter((subscribers) -> subscribers > 0)
+								.switchIfEmpty(this.subscriptionDelay.next()))
+				.subscribe();
```

Comment:
I think the proper way would be to have a `subscribeTo()` to return `Mono<Void>` instead to let that caller to subscribe or compose further its reactive flow.
This way we would defer an emission from this source publisher as late as possible.

But I'd like to revise that stuff in a separate PR.

Thanks